vllm/v1/structured_output/backend_outlines.py

```diff
@@ -0,0 +1,310 @@
+# SPDX-License-Identifier: Apache-2.0
+from __future__ import annotations
+
+import ast
+import json
+import re
+import sys
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING
+
+import torch
+
+from vllm.config import VllmConfig
+from vllm.model_executor.guided_decoding.outlines_logits_processors import (
+    OutlinesVocabulary, get_cache, get_vocabulary)
+from vllm.sampling_params import SamplingParams
+from vllm.transformers_utils.tokenizer_group import init_tokenizer_from_configs
+from vllm.utils import LazyLoader
+from vllm.v1.structured_output.backend_types import (StructuredOutputBackend,
+                                                     StructuredOutputGrammar,
+                                                     StructuredOutputOptions)
+
+if TYPE_CHECKING:
+    import outlines_core as oc
+else:
+    oc = LazyLoader("oc", globals(), "outlines_core")
+
+# Python 3.11+ sre_parse and sre_constants
+# are deprecated, so we must import them from re
+if sys.version_info >= (3, 11):
+    from re import _constants as sre_constants  # type: ignore[attr-defined]
+    from re import _parser as sre_parse  # type: ignore[attr-defined]
+else:
+    import sre_constants
+    import sre_parse
+
+
+class OutlinesBackend(StructuredOutputBackend):
```

Comment:
I have some slight interface changes from https://github.com/vllm-project/vllm/pull/16577/files#diff-67e43a71961e7dba748b644944979e3f359e9b674eb30bfd836b66a75990a9c1

Can you update this as well? thanks.